### ISSUE TITLE
Run Name by default for the execution report files

### ISSUE THREAD

**Issue by edgano** [nextflow-io/nextflow#630]
The main idea is to have the output files (report/trace/...) of each run with the unique name given by nextflow.
It will be more readable instead of the trace.txt, trace.txt.1 ,...
![screenshot from 2018-03-07 18-42-40](https://user-images.githubusercontent.com/4262262/37108425-5799788a-2237-11e8-8f4d-90cd166e6b49.png)
The idea is to have "by default" the name **marvelous_meucci**

**Comment by pditommaso** [comment]
Make sense, I'd propose to have those files named as `trace.<run-name>.txt`. Still not sure if this should be the default (it could break existing application relying on the `trace.txt` name). 

Maybe at least at the beginning it could an opt-in feature, adding a config setting `autoNames`, or something like that, in the config file.

**Comment by pditommaso** [comment]
Any feedback from @skptic @emi80 @ewels ?

**Comment by ewels** [comment]
I like the idea! Especially if it uses whatever the `workflow.name` is, so that if you use `-name MyRun` when running the pipeline the reports get to be `report.MyRun.html`. We would find this very helpful (already do the same for MultiQC reports but hadn't figured out how to do it with Nextflow reports).

NB: Would it be possible to add a nextflow specific string at the same time? eg. instead of just `report`, having `nf_report.MyRun.html` or something.. (no reason for requesting this other than a deep-set dislike of generic filenames caused by programming MultiQC modules).

**Comment by ewels** [comment]
ps. If possible, it would be good to have this functionality whilst also being able to define where the file should be stored. For example, we currently set the report filename as follows:

```groovy
file = "${params.outdir}/pipeline_info/pipeline_trace.txt"
```

I like having the reports ending up in the output directory instead of the working directory, but presumably it will be difficult to get the workflow name into here as well..?

**Comment by pditommaso** [comment]
Not sure, but this may work already.

**Comment by jmarti-n1h** [comment]
Hi,

I may be needing this feature. Any updates? Any work around?

JMC

**Comment by bentsherman** [comment]
It might work already. Have you tried to make the file name dynamic based on the run name? Something like...
```groovy
report.file = "${workflow.runName}.report.html"
```

**Comment by ewels** [comment]
This is exactly what we do in all @nf-core workflows. See the [template](https://github.com/nf-core/tools/blob/13f331ee139a62bda75244cc125d9d5fa308a7a9/nf_core/pipeline-template/nextflow.config#L218-L234):

```nextflow
def trace_timestamp = new java.util.Date().format( 'yyyy-MM-dd_HH-mm-ss')
report {
    enabled = true
    file    = "${params.outdir}/pipeline_info/execution_report_${trace_timestamp}.html"
}
```

**Comment by bentsherman** [comment]
Yeah, I'm just wondering if the `workflow` variable is available in the config

**Comment by jmarti-n1h** [comment]
No, `worflow` or `workDir` seemed to be unavailable in `conf`. @ewels , is `${params.outdir}` defined in `main.nf` or somewhere else?

**Comment by ewels** [comment]
It's defined further up in the `nextflow.config` file. The squiggly brackets (`{}`) defer execution, so the config is fully resolved before the value is set.

**Comment by bentsherman** [comment]
@zhilizheng confirmed that `workflow` isn't available in the config file (see #4145). You have to use a closure to access that variable. There is actually a PR #3651 to support closures for these config fields, let's continue the discussion there.
